### PR TITLE
Backport GitHub Actions integration to stable-4.11

### PR DESCRIPTION
**Pull request by fingolfin**
Also update the README.md badges, which were still pointing to the master branch (this also lead to PR #4173). In the future. "update badges in README.md" should be part of the instructions for creating a new stable-X.Y branch; hence I made gap-system/gap-distribution#116

### CODE REVIEW

File: src/gasman.c [hunk 1]
```diff
@@ -1815,10 +1815,11 @@ static NOINLINE void GenStackFuncBags(void)
...
 
-    /* mark from registers, dirty dirty hack                               */
-    for ( p = (Bag*)((void*)RegsBags);
-          p < (Bag*)((void*)RegsBags)+sizeof(RegsBags)/sizeof(Bag);
-          p++ )
+    // mark from registers, dirty dirty hack: we treat the jmp_buf as a
+    // sequence of Bag values. Note that sizeof(jmp_buf) need not be a
+    // multiple of sizeof(Bag), hence the end condition looks slightly.
+    // unusual.
+    for (p = (Bag *)RegsBags; p + 1 <= (Bag *)(RegsBags + 1); p++)
```
Comment (wilfwilson) [inline src/gasman.c:1822]: Is this related to backporting GitHub actions? I see that this is how the code already looks in the `master` branch, I don't object to making the change to `stable-4.11`, I'm just curious why it's here.

**Review by wilfwilson — APPROVED** [review]

File: src/gasman.c (hunk 1 above)
Comment (fingolfin) [reply src/gasman.c:1822]: I addressed this in the commit message (but forgot to update the PR description):
> The tweak in src/gasman.c resolves a compiler warning that'd cause a CI failure (as we compile with -Werror)

Comment (wilfwilson) [reply src/gasman.c:1822]: Thanks for the explanation. 🙂 I normally look at the commit messages, but I forgot this time.
